msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/AnonymousManagerImpl.java

```diff
@@ -0,0 +1,198 @@
+package org.sakaiproject.component.app.messageforums;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.hibernate.Hibernate;
+import org.hibernate.HibernateException;
+import org.hibernate.Query;
+import org.hibernate.Session;
+import org.hibernate.exception.ConstraintViolationException;
+import org.sakaiproject.api.app.messageforums.AnonymousManager;
+import org.sakaiproject.api.app.messageforums.AnonymousMapping;
+import org.sakaiproject.component.app.messageforums.dao.hibernate.AnonymousMappingImpl;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.orm.hibernate3.HibernateCallback;
+import org.springframework.orm.hibernate3.support.HibernateDaoSupport;
+
+public class AnonymousManagerImpl extends HibernateDaoSupport implements AnonymousManager
+{
+	private static final Log LOG = LogFactory.getLog(AnonymousManagerImpl.class);
+
+	private final String ANON_ID_PADDING = "000000";
+
+	private final String QUERY_BY_SITE = "findMappingsBySite";
+	private final String QUERY_BY_SITE_AND_USERS = "findMappingsBySiteAndUsers";
+
+	private final int MAX_IN_CLAUSE_SIZE = 1000;
+
+	public String getAnonId(final String siteId, final String userId)
+	{
+		// Query for a single user in the site
+		List<AnonymousMapping> results = findMappingsBySiteAndUsers(siteId, Collections.singletonList(userId));
+
+		if (CollectionUtils.isEmpty(results))
+		{
+			return null;
+		}
+
+		return results.get(0).getAnonId();
+	}
+
+	public String getOrCreateAnonId(final String siteId, final String userId)
+	{
+		return getOrCreateUserIdAnonIdMap(siteId, Collections.singletonList(userId)).get(userId);
+	}
+
+	private List<AnonymousMapping> findMappingsBySiteAndUsers(final String siteId, final List<String> userIds)
+	{
+		if (CollectionUtils.isEmpty(userIds))
+		{
+			return Collections.emptyList();
+		}
+
+		HibernateCallback<List<AnonymousMapping>> hcb = new HibernateCallback<List<AnonymousMapping>>()
```

Comment:
This is the compilation error that threw me off and lead me to assume that diamond operators didn't compile in msgcntr:
cannot infer type arguments for org.springframework.orm.hibernate3.HibernateCallback<T>
[ERROR] reason: cannot use '<>' with anonymous inner classes

So, no anonymous classes, but I'm applying them everywhere else in the refactor